monsteer/plugin/spikeReport.cpp

```diff
@@ -49,19 +50,26 @@ namespace
 const uint32_t INTERNAL_TIMEOUT = 500;
 }
 
+URI toHostAndPort( const URI& uri )
+{
+    URI out;
+    out.setHost( uri.getHost( ));
+    out.setPort( uri.getPort( ));
+    return out;
+}
+
 SpikeReport::SpikeReport( const brion::SpikeReportInitData& pluginData )
-    : _uri( pluginData.getURI( ))
+    : _uri( toHostAndPort( pluginData.getURI( )))
```

Comment:
True, this isn't very elegant to degenerate to a string representation in this specific case. We could also have a Servus::HostPort class to keep more semantic value.
One cause of the problem here is that the URI is used both for plugin selection and as a ZMQ connection info which forces us to remove the scheme. This might also be annoying if we want to use different ZMQ protocols in the future (e.g. inproc:// instead of tcp://).
Maybe as you suggest we can keep it like this for this commit and agree on a more generic solution later.